### PR TITLE
Implemented "Select All" Toggle for Filter from factor

### LINKED ISSUE #7545: "Select all" button text in the Filter From Factors sub dialog does not toggle to "Unselect all" after click it
This button should behave like the one in the define filter sub dialog.

### PR DESCRIPTION
**Pull request by derekagorhom**
Fixes #7545 
Implemented the toggle for `select all` on the filter from factor sub dialogue.

@Patowhiz, @N-thony This PR is ready for review.

### CODE REVIEW

File: instat/sdgFiltersFromFactor.vb [hunk 1]
```diff
@@ -70,29 +70,43 @@ Public Class sdgFiltersFromFactor
...
-            cmdSelectAll.Text = "Deselect All Levels"
-            cmdSelectAll.FlatStyle = FlatStyle.Flat
+            cmdToggleSelectAll.Text = "Deselect All Levels"
+            cmdToggleSelectAll.FlatStyle = FlatStyle.Flat
         Else
-            cmdSelectAll.Text = "Select All Levels"
-            cmdSelectAll.FlatStyle = FlatStyle.Popup
+            cmdToggleSelectAll.Text = "Select All Levels"
+            cmdToggleSelectAll.FlatStyle = FlatStyle.Popup
         End If
```
Comment (Patowhiz) [inline instat/sdgFiltersFromFactor.vb:97]: @derekagorhom I think the toggling functionality here is no longer needed. So these code lines can just be removed.
Whenever the receiver changes it's value, the factor control will also change it's value. So `ucrFactorLevels.ControlValueChanged` will be raised. Making the toggling code lines here redundant.

File: instat/sdgFiltersFromFactor.vb [hunk 2]
```diff
@@ -55,7 +55,7 @@ Public Class sdgFiltersFromFactor
                                                   dctParamAndColNames:=dctParamAndColNames,
                                                   hiddenColNames:={ucrFactor.DefaultColumnNames.Level},
                                                   bIncludeNALevel:=False)
-        cmdSelectAll.Enabled = False
+        cmdToggleSelectAll.Enabled = False
     End Sub
```
Comment (Patowhiz) [inline instat/sdgFiltersFromFactor.vb:59]: @derekagorhom I also don't think we need this code line 58. Check if removing changes the implementation.

Comment (derekagorhom) [reply instat/sdgFiltersFromFactor.vb:58]: @Patowhiz i have made the changes you requested, can you review them again, thanks.

**Comment by derekagorhom** [comment]
@Patowhiz can you review this again

**Comment by N-thony** [comment]
> @Patowhiz can you review this again

@Patowhiz can you review this? Thanks

**Comment by Patowhiz** [comment]
@derekagorhom thanks for this.
I looked at the code and made changes to it in PR #7859. 
I'm not quite sure what the design changes are. So I didn't include them.
I hope that's okay with you.
Kindly review it and we can discuss.
Thanks.

**Comment by Patowhiz** [comment]
I suggest this to now be closed

**Comment by derekagorhom** [comment]
> @derekagorhom thanks for this. I looked at the code and made changes to it in PR #7859. I'm not quite sure what the design changes are. So I didn't include them. I hope that's okay with you. Kindly review it and we can discuss. Thanks.

Thanks @Patowhiz i will have a look at it

**Pull request closed without merging**